tests/interactive_broker_version.py

```diff
@@ -33,229 +25,16 @@ def version_as_number(version):
-    print('# Using client security.protocol=%s' % security_protocol)
-    all_listeners = (
-        ','.join(
-            cluster.get_all(
-                'listeners',
-                '',
-                KafkaBrokerApp))).split(',')
-    bootstrap_servers = ','.join(
-        [x for x in all_listeners if x.startswith(security_protocol)])
-    os.write(fd, ('bootstrap.servers=%s\n' %
-             bootstrap_servers).encode('ascii'))
-    os.write(fd, ('security.protocol=%s\n' %
-             security_protocol).encode('ascii'))
-    os.close(fd)
-
-    if deploy:
-        print('# Deploying cluster')
-        cluster.deploy()
-    else:
-        print('# Not deploying')
-
-    print('# Starting cluster, instance path %s' % cluster.instance_path())
-    cluster.start()
-
-    print('# Waiting for brokers to come up')
-
-    if not cluster.wait_operational(30):
-        cluster.stop(force=True)
-        raise Exception('Cluster %s did not go operational, see logs in %s/%s' %  # noqa: E501
-                        (cluster.name, cluster.root_path, cluster.instance))
-
-    print('# Connect to cluster with bootstrap.servers %s' % bootstrap_servers)
-
-    cmd_env['KAFKA_PATH'] = brokers[0].conf.get('destdir')
-    cmd_env['RDKAFKA_TEST_CONF'] = test_conf_file
-    cmd_env['ZK_ADDRESS'] = zk_address
-    cmd_env['BROKERS'] = bootstrap_servers
-    cmd_env['TEST_KAFKA_VERSION'] = version
-    cmd_env['TRIVUP_ROOT'] = cluster.instance_path()
-    cmd_env['TEST_SCENARIO'] = scenario
-
-    # Provide a HTTPS REST endpoint for the HTTP client tests.
-    cmd_env['RD_UT_HTTP_URL'] = 'https://jsonplaceholder.typicode.com/users'
-
-    # Per broker env vars
-    for b in [x for x in cluster.apps if isinstance(x, KafkaBrokerApp)]:
-        cmd_env['BROKER_ADDRESS_%d' % b.appid] = \
-            ','.join([x for x in b.conf['listeners'].split(
-                ',') if x.startswith(security_protocol)])
-        # Add each broker pid as an env so they can be killed indivdidually.
-        cmd_env['BROKER_PID_%d' % b.appid] = str(b.proc.pid)
-        # JMX port, if available
-        jmx_port = b.conf.get('jmx_port', None)
-        if jmx_port is not None:
-            cmd_env['BROKER_JMX_PORT_%d' % b.appid] = str(jmx_port)
-
-    if not cmd:
-        cmd_env['PS1'] = '[TRIVUP:%s@%s] \\u@\\h:\\w$ ' % (
-            cluster.name, version)
-        cmd = 'bash --rcfile <(cat ~/.bashrc)'
-
-    ret = True
-
-    for i in range(0, exec_cnt):
-        retcode = subprocess.call(
-            cmd,
-            env=cmd_env,
-            shell=True,
-            executable='/bin/bash')
-        if retcode != 0:
-            print('# Command failed with returncode %d: %s' % (retcode, cmd))
-            ret = False
-
-    try:
-        os.remove(test_conf_file)
-    except BaseException:
-        pass
-
-    cluster.stop(force=True)
-
-    cluster.cleanup(keeptypes=['log'])
-    return ret
+    conf['test_mode'] = 'bash'
```

Comment:
This should be part of the test config instead of hardcoding.

Reply:
in interactive_broker_version it was always in bash mode, I've kept existing behavior. Because it's interactive, for other modes there is `broker_version_tests.py`

Reply:
But bash might not be available in MacOS. We should give option to the user to choose it.

Reply:
Before was calling bash too
`        retcode = subprocess.call(
            cmd,
            env=cmd_env,
            shell=True,
            executable='/bin/bash')`

Reply:
`run-test.sh` is requiring bash too, it's requisite for developing with librdkafka already

Reply:
Let it be like this for now. We should have zsh support as well in the future though.